Website/index.js

```diff
@@ -0,0 +1,474 @@
+        $(v).hide();
+    });
+
+    function today(date) {
+        var inputDate = new Date(date);
+        var todaysDate = new Date();
+        if (inputDate.setHours(0, 0, 0, 0) == todaysDate.setHours(0, 0, 0, 0)) {
+            return true
+        } else {
+            return false
+        }
+    }
+
+    function update_block_arrangement() {
+        var startTime = '06:00:00';
+        var endTime = '08:00:00';
+
+        currentDate = new Date();
+
+        startDate = new Date(currentDate.getTime());
+        startDate.setHours(startTime.split(":")[0]);
+        startDate.setMinutes(startTime.split(":")[1]);
+        startDate.setSeconds(startTime.split(":")[2]);
+
+        endDate = new Date(currentDate.getTime());
+        endDate.setHours(endTime.split(":")[0]);
+        endDate.setMinutes(endTime.split(":")[1]);
+        endDate.setSeconds(endTime.split(":")[2]);
+
+
+        valid = startDate < currentDate && endDate > currentDate
+        if (valid && today(String($("#start").val()))) {
+            $("#recess_duty_block").insertAfter("#morning_duty_block");
+        } else {
+            $("#recess_duty_block").insertBefore("#morning_duty_block");
+        }
+
+    }
+
+    function list_to_string(list) {
+        if (list == "") {
+            return "-"
+        } else {
+            return list.join(", ");
+        }
+    }
+
+    function update_morning_duty() {
+        morning_duty_data = jsonData["constants"][jsonData["presets"][jsonData["dates"][String($("#start").val())]]["constant_morning_duty"]];
+        $("#morning_main_gate").text(list_to_string(morning_duty_data["main_gate"]));
+        $("#morning_late_coming").text(list_to_string(morning_duty_data["late_coming"]));
+        $("#morning_canteen").text(list_to_string(morning_duty_data["canteen"]));
+        $("#morning_stairs_to_hall").text(list_to_string(morning_duty_data["stairs_to_hall"]));
+        $("#morning_side_gate").text(list_to_string(morning_duty_data["side_gate"]));
+        $("#morning_lift").text(list_to_string(morning_duty_data["lift"]));
+        $("#morning_morning_patrol").text(list_to_string(morning_duty_data["morning_patrol"]));
+        $("#morning_hall").text(list_to_string(morning_duty_data["hall"]));
+
+    }
+    function update_flag_duty() {
+        data = flagData[String($("#start").val())]
+
+        if (data == null) {
+            $("#morning_flag_singapore").text("Data is unavailable");
+            $("#morning_flag_school").text("Data is unavailable");
+        } else {
+            $("#morning_flag_singapore").text(data["singapore_flag"]);
+            $("#morning_flag_school").text(data["school_flag"]);
+        }
+    }
+
+    function update_date() {
+        var dtt = new Date(String($("#start").val()));
+        var days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Davianlols6/SC-Duty-List/pullRequest?prid=8386818)
```suggestion
    var days = ["Sunday", 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
```